src/Microsoft.ML.Data/Scorers/PredictionTransformer.cs

```diff
@@ -769,6 +770,37 @@ internal static object Create(IHostEnvironment env, ModelLoadContext ctx, IHost
 
             return genericInstance;
         }
+
+        internal static Type GetLoadType(Type modelType)
+        {
+            var att = modelType.GetCustomAttribute(typeof(PredictionTransformerLoadType)) as PredictionTransformerLoadType;
+            if (att != null)
+            {
+                if (att.LoadType.IsGenericType && att.LoadType.GetGenericArguments().Length == modelType.GetGenericArguments().Length)
+                {
+                    // This assumes that if att.LoadType and modelType have the same number of type parameters
+                    // Then they should get the same type parameters.
+                    // This is the case for CalibratedModelParametersBase and its children generic clases.
+                    // But might break if other classes begin using the PredictionTransformerLoadtype attribute in the future.
+                    Type[] typeArguments = modelType.GetGenericArguments();
+                    Type genericType = att.LoadType;
+                    return genericType.MakeGenericType(typeArguments);
+                }
+            }
+
+            return modelType;
+        }
+    }
+
+    [AttributeUsage(AttributeTargets.Class)]
+    internal class PredictionTransformerLoadType : Attribute
+    {
+        internal Type LoadType;
```

Comment:
Use a property instead of a field.

```C#
internal Type LoadType { get; }
```

Reply:
I will change this asap. I just wanted to ask why is it better to use a property in this case? I wonder so that I can better choose between properties and fields in the future. Thanks!

Reply:
In general, if you are exposing a value outside of a class, use a property. The benefits are:

1. You can disallow others from changing the value. Fields are settable (like yours above).
2. You can intercept the call, if necessary. For example, validate a value being set is non-null. Or you can lazy-load the property if it isn't already initialized.
3. It provides more flexibility in the future for public APIs. If you ship a public field in a public API you cannot change it to a property later.

In your current code, you are allowing anyone else in the assembly to change this value (whether they intended to or made a mistake).

It is a good habit to get into to always use properties for non-private instance data.

See https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/field for more info.